spec/haskell/Makefile

```diff
@@ -74,6 +74,7 @@ build-riscv: sandbox $(BOOT_FILES)
 # cabal package database is missing, the build will fail.
 
 .stack-work:
+	mkdir ~/.stack
```

Comment:
Just curious, is this needed for the changes in this PR specifically or for something else?

Reply:
This is needed because on github, the user home (`~`) is not owned by the CI runner, and `stack` will refuse to create the directory `~/.stack` when that is the case.